backend/mem/file.go

```diff
@@ -0,0 +1,469 @@
+package mem
+
+import (
+	"errors"
+	"github.com/c2fo/vfs/v4"
+	"github.com/c2fo/vfs/v4/utils"
+	"io"
+	"path"
+	"time"
+)
+
+//File implements vfs.File interface for os fs.
+type File struct {
+	exists    bool
+	timeStamp time.Time
+	isOpen    bool
+	isRef     bool         //has it been referenced before?
+	privSlice []byte       //the file contents
+	Filename  string       //the base name of the file
+	cursor    int          //the index that the buffer (privSlice) is at
+	location  vfs.Location //the location that the file exists on
+}
+
+/*		******* Error Functions *******		*/
+func doesNotExist() error {
+	return errors.New("This file does not exist!")
+}
+func copyFail() error {
+	return errors.New("This file was not successfully copied")
+}
+
+func copyFailNil() error {
+	return errors.New("The target file passed in was nil")
+}
+
+func deleteError() error {
+	return errors.New("Deletion was unsuccessful")
+}
+
+func moveToLocationError() error {
+	return errors.New("Move to location unexpectedly failed")
+}
+func writeError() error {
+	return errors.New("Unexpected Write Error")
+}
+
+func seekError() error {
+	return errors.New("Seek could not complete the desired call")
+}
+
+/*		******************		*/
+
+//Close imitates io.Closer by resetting the cursor and setting a boolean
+func (f *File) Close() error {
+	if !f.isRef {
+		// Do nothing on files that were never referenced
+		return nil
+	}
+
+	f.isOpen = false
+	f.cursor = 0
+	return nil
+}
+
+//Read implements the io.Reader interface.  Returns number of bytes read and potential errors
+func (f *File) Read(p []byte) (n int, err error) {
+	//if file exists:
+	if f.isOpen == false {
+		f.isOpen = true
+	}
+	existence, eerr := f.Exists()
+	if !existence {
+		return 0, eerr
+	}
+	f.isRef = true
+	length := len(p)
+	if length == 0 { //length of byte slice is zero, just return 0 and nil
+		return 0, nil
+	}
+
+	if f.cursor == len(f.privSlice) {
+		return 0, io.EOF
+	}
+	for i := f.cursor; i < length; i++ {
+		f.cursor++
+		if i >= length || i >= len(f.privSlice) {
+			break
+		}
+		if i == length {
+			break
+		}
+		p[i] = f.privSlice[i]
+	}
+	f.timeStamp = time.Now()
```

Comment:
why are we modifying the time on a read?  If it was last accessed, that would make sense but we only care about last modified.